server/sonar-server/src/main/java/org/sonar/server/computation/sqale/SqaleRatingSettings.java

```diff
@@ -0,0 +1,152 @@
+ */
+
+package org.sonar.server.computation.sqale;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import org.sonar.api.config.Settings;
+
+import static java.lang.String.format;
+import static org.sonar.api.CoreProperties.DEVELOPMENT_COST;
+import static org.sonar.api.CoreProperties.LANGUAGE_SPECIFIC_PARAMETERS;
+import static org.sonar.api.CoreProperties.LANGUAGE_SPECIFIC_PARAMETERS_LANGUAGE_KEY;
+import static org.sonar.api.CoreProperties.LANGUAGE_SPECIFIC_PARAMETERS_MAN_DAYS_KEY;
+import static org.sonar.api.CoreProperties.LANGUAGE_SPECIFIC_PARAMETERS_SIZE_METRIC_KEY;
+import static org.sonar.api.CoreProperties.RATING_GRID;
+import static org.sonar.api.CoreProperties.RATING_GRID_DEF_VALUES;
+import static org.sonar.api.CoreProperties.SIZE_METRIC;
+
+public class SqaleRatingSettings {
+
+  private final Settings settings;
+  private final Map<String, LanguageSpecificConfiguration> languageSpecificConfigurationByLanguageKey;
+
+  public SqaleRatingSettings(Settings settings) {
+    this.settings = settings;
+    this.languageSpecificConfigurationByLanguageKey = buildLanguageSpecificConfigurationByLanguageKey(settings);
+  }
+
+  private static Map<String, LanguageSpecificConfiguration> buildLanguageSpecificConfigurationByLanguageKey(Settings settings) {
+    ImmutableMap.Builder<String, LanguageSpecificConfiguration> builder = ImmutableMap.builder();
+    String[] languageConfigIndexes = settings.getStringArray(LANGUAGE_SPECIFIC_PARAMETERS);
+    for (String languageConfigIndex : languageConfigIndexes) {
+      String languagePropertyKey = LANGUAGE_SPECIFIC_PARAMETERS + "." + languageConfigIndex + "." + LANGUAGE_SPECIFIC_PARAMETERS_LANGUAGE_KEY;
+      builder.put(settings.getString(languagePropertyKey), LanguageSpecificConfiguration.create(settings, languageConfigIndex));
+    }
+    return builder.build();
+  }
+
+  public double[] getRatingGrid() {
+    try {
+      String[] ratingGrades = settings.getStringArray(RATING_GRID);
+      double[] grid = new double[4];
+      for (int i = 0; i < 4; i++) {
+        grid[i] = Double.parseDouble(ratingGrades[i]);
+      }
+      return grid;
+    } catch (Exception e) {
+      throw new IllegalArgumentException("The SQALE rating grid is incorrect. Expected something similar to '"
+        + RATING_GRID_DEF_VALUES + "' and got '"
+        + settings.getString(RATING_GRID) + "'", e);
+    }
+  }
+
+  public long getDevCost(@Nullable String languageKey) {
+    if (languageKey != null) {
+      try {
+        LanguageSpecificConfiguration languageSpecificConfig = getSpecificParametersForLanguage(languageKey);
+        if (languageSpecificConfig != null && languageSpecificConfig.getManDays() != null) {
+          return Long.parseLong(languageSpecificConfig.getManDays());
+        }
+      }
+      catch (NumberFormatException e) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this "catch" on the same line that the previous closing curly brace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARightCurlyBraceSameLineAsNextBlockCheck)